stages/org.osbuild.dmverity

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python3
+import os
+import subprocess
+import sys
+import tempfile
+import osbuild.api
+
+
+def open_dmverity(blocksize, partlabel, datapart, hashpart, mountpoint):
+    hash = subprocess.run(["/usr/sbin/veritysetup",
+                            "format", datapart, hashpart,
+                            "--data-block-size", f"{blocksize}"],
+                           check=True, capture_output=True)
+    for line in hash.stdout.decode().splitlines():
+      if line.startswith("Root hash:"):
+        hash = line.split(":", 1)[1].strip()
+        break
+
+    subprocess.run(["/usr/sbin/veritysetup",
+                    "open", datapart, partlabel, hashpart, hash],
+                   check=True)
+    subprocess.run(["mount", "-o", "ro", f"/dev/mapper/{partlabel}", mountpoint],
+                   check=True)
+    return hash
+
+
+def close_dmverity(partlabel, mountpoint):
+    subprocess.run(["umount", mountpoint], check=False)
+    subprocess.run(["/usr/sbin/veritysetup", "close", partlabel], check=False)
+
+
+def main(tree, paths, devices, options):
+    blocksize = options.get("blocksize", 512)
+
+    disk = os.path.join(paths["devices"], devices["disk"]["path"])
+    partition = f"{disk}p{options["partition"]}"
+    hash_partition = f"{disk}p{options["hash_partition"]}"
+
+    label = options["label"]
+    hashfile = os.path.join(tree, f"{label}_hash")
+
+    with tempfile.TemporaryDirectory() as root:
+      try:
+        hash = open_dmverity(blocksize, label, partition, hash_partition, root)
```

Comment:
*if* the "veritysetup open"/"mount" is not actually required this can be simplfiied and the TemporaryDirectory() can be removed and the function should probably be renamed to somehting like "setup_dmverity" or "format_dmverity" or something.